src/quo2/components/permission_tag.cljs

```diff
@@ -0,0 +1,125 @@
+(ns quo2.components.permission-tag
+  (:require [status-im.ui.components.react :as react]
+            [quo2.components.text :as text]
+            [quo2.foundations.colors :as colors]
+            [quo2.components.icon :as icons]
+            [status-im.i18n.i18n :as i18n]))
+
+(defn permission-tag-styles [color size]
+  {:flex-direction   :row
+   :border-radius    200
+   :height           size
+   :align-items      :center
+   :justify-content  :flex-end
+   :background-color color})
+
+(defn outer-resource-container [size]
+  {:background-color (colors/theme-colors
+                      colors/neutral-10
+                      colors/neutral-80)
+   :border-radius    (case size 32 32 24 22)
+   :width            (case size 32 32 24 22)
+   :height           (case size 32 32 24 22)
+   :margin-left      (case size 32 -12 24 -8)
+   :align-items      :center
+   :justify-content  :center})
+
+(defn extra-count-styles [size]
+  {:background-color (colors/theme-colors
+                      colors/neutral-20
+                      colors/neutral-70)
+   :height           (case size 32 28 24 20)
+   :width            (case size 32 28 24 20)
+   :border-radius    size
+   :justify-content  :center
+   :align-items      :center})
+
+(defn tag-resources [tokens size]
+  (let [tkns (take (if (> (count tokens) 3) 2 3) tokens)]
+    (for [{:keys [token-icon]} tkns]
+      ^{:key token-icon}
+      [react/view {:flex-direction :row}
+       [react/view (outer-resource-container size)
+        [react/image {:source token-icon
+                      :style  {:height        (case size 32 28 24 20)
+                               :width         (case size 32 28 24 20)
+                               :border-radius size}}]]])))
+
+(defn tag-extra-count [tokens size]
+  (let [taken-tkns (take (if (> (count tokens) 3) 2 3) tokens)
```

Comment:
IMO dropping letters from variable names offers no value. Here it's quite obvious what's implied but I think it's better to be implicit and use full words.
e.g `taken-tkns` -> `taken-tokens` - reduces the chance of another dev being confused.